src/strnum.F90

```diff
@@ -0,0 +1,73 @@
+!> @file
+!> @brief Decode an integer from a character string.
+!>
+!> @author J. Woollen @date 1994-01-06
+
+!> This subroutine decodes an integer from a character string.  The
+!> string may contain leading or trailing blanks, but otherwise should
+!> contain only digits and an (optional) leading sign ('+' or '-')
+!> character. 
+!>
+!> If the string contains all blank characters, then num is returned
+!> with a value of 0.
+!>
+!> @param[in]  str -- character*(*): String
+!> @param[out] num -- integer: Value decoded from str
+!> @param[out] iret -- return code:
+!>   0 = success
+!>  -1 = string contained one or more illegal characters
+!>
+!> @author J. Woollen @date 1994-01-06
+recursive subroutine strnum(str,num,iret)
+  use modv_im8b
+
+  implicit none
+
+  character*(*), intent(in) :: str
+
+  integer, intent(out) ::  num, iret
+
+  character str2*40, fmt*8
+
+  integer lens
+
+!-----------------------------------------------------------------------
+!-----------------------------------------------------------------------
+
+! Check for I8 integers.
+
+  if (im8b) then
+     im8b = .false.
+
+     call strnum ( str, num, iret )
+     call x48 ( num, num, 1 )
+     call x48 ( iret, iret, 1 )
+
+     im8b = .true.
+     return
+  end if
+
+  iret = 0
+
+! Check for blank input string.
+
+  if ( str .eq. ' ' ) then
+    num = 0
+    return
+  end if
+
+! Verify that the string contains all legal characters.
+
+  call strsuc ( str, str2, lens )
+  if ( verify ( str2(1:lens), "0123456789+-" ) .ne. 0 ) then
+    iret = -1
+    return
+  end if
+
+! Decode the integer from the string.
+
+  write ( fmt, '(''(I'',I2,'')'')' ) lens
+  read ( str2(1:lens), fmt ) num
```

Comment:
If non-numeric characters are in the string, won't this fail? In which case, why try to detect non-numeric characters? Why not just try to read and if there is a failure, report a failure?

Reply:
OK, I just saw your #409 (sorry, it's hard keeping track of all of these overlapping conversations and PR's flying around ;-)

So I take it you would indeed prefer that I just modify the new strnum.F90 to add an iostat= check to the read and remove the verify check, and I see where you've already included some very thorough strnum testing in test_misc.F90, so I'll go ahead and remove my less thorough tests from intest7.F90 as well.